src/System.IO.FileSystem/src/System/IO/File.cs

```diff
@@ -386,6 +403,76 @@ private static String InternalReadAllText(String path, Encoding encoding)
                 return sr.ReadToEnd();
         }
 
+
+        // If we use the path-taking constructors we will not have FileOptions.Asynchronous set and
+        // we will have asynchronous file access faked by the thread pool. We want the real thing.
+        private static StreamReader AsyncStreamReader(string path, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
+                path, FileMode.Open, FileAccess.Read, FileShare.Read, DefaultBufferSize,
+                FileOptions.Asynchronous | FileOptions.SequentialScan);
+
+            return new StreamReader(stream, encoding, true);
+        }
+
+        private static StreamWriter AsyncStreamWriter(string path, bool append, Encoding encoding)
+        {
+            FileStream stream = new FileStream(
+                path, append ? FileMode.Append : FileMode.Create, FileAccess.Write, FileShare.Read, DefaultBufferSize,
+                FileOptions.Asynchronous | FileOptions.SequentialScan);
+
+            return new StreamWriter(stream, encoding);
+        }
+
+        public static Task<string> ReadAllTextAsync(
+            string path, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            return ReadAllTextAsync(path, Encoding.UTF8, cancellationToken);
+        }
+
+        [System.Security.SecuritySafeCritical] // auto-generated
```

Comment:
That slipped in, but I was going to ask anyway, what, if any security attributes should be here. I understood .NET Core didn't use these, yes?